### ISSUE TITLE
can't re-link a slack space after dropping then re-adding integration

### ISSUE THREAD

**Issue by mavigro** [matrix-org/matrix-appservice-slack#321]
Very similar to issue #308  that was closed without a final resolution.
1. initially integrated a slack space called biz-tech-team@slack.com that was working
2. dropped the integration in slack and cleaned-up references to the bot user in the matrix room
3. tried to re-add the integration but as show in the attachment the channel list refuses to expand
4. tried integrating a 2nd slack space digitalcurrencycrew@slack.com to the same room which works
5. in between the two integrations I accidentally un-joined myself from the original room crmtalk:matrix.org which I can no longer access (associated with community ohana:matrix.org) so I re-created the room with the same name but that didn't help the first integration.
6. created a 2nd room biztechteam:matrix.org which like the original room cannot be integrated with the the first slack space biz-tech-team@slack.com but was successfully integrated with the 2nd slack space digitalcurrencycrew@slack.com so it seems something got corrupted with the integration of the first slack space.
![slack-integration-issue](https://user-images.githubusercontent.com/29416405/68770362-fb540200-05f3-11ea-9615-e486edfa88de.png)

**Comment by auscompgeek** [comment]
> dropped the integration in slack

So you deleted the integration on the Slack side (revoking the Riot bridge's permissions) but didn't log out on the Riot integration manager side?

**Comment by mavigro** [comment]
yes, that's quite possible as I dropped and attempted to re-add the integration numerous times. Also at one point while it was working I renamed the bot in slack but don't recall that having a negative effect.

**Comment by mavigro** [comment]
Eureka! with the appservice installed in the non-responding Slack I logged-out of the non-integration in Riot THEN dropped and re-added the appservice to the non-responding Slack and now the channel list expands in Riot.

I'm certain I did that umpteen times but there's one additional observation that may lend insight, after a Matrix HQ member advised on how to drop the orphaned (originally integrated) Room from the Community before doing so I tried once more to access the room and noticed this time the error "Failed to join Room - No Available Servers" vs last night I recall a different error indicating needing  an invite so it seems some state with the original room (and perhaps the original integration with the inaccessible Slack) got purged.

So after re-integrating the previously stuck Slack now when clicking the 4-square integration icon in the room the modal box comes back with "Existing Integrations" which wasn't happening before and the integration works perfectly.

Kudos to everyone that had a hand in creating this amazing appservice and thanks to those that reached-out to advise as I worked through this issue. Thx
